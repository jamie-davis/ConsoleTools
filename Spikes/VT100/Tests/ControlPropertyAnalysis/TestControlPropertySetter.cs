using System.Linq;
using TestConsoleLib;
using TestConsoleLib.Testing;
using VT100.ControlPropertyAnalysis;
using VT100.FullScreen;
using Xunit;

namespace VT100.Tests.ControlPropertyAnalysis
{
    public class TestControlPropertySetter
    {
        [Fact]
        public void PropertiesAreExtracted()
        {
            //Arrange
            var props = new []
            {
                new PropertySetting<VtColour>("Background", VtColour.Black),
                new PropertySetting<VtColour>("Foreground", VtColour.White),
            };
            var propClass = new ScreenProps();

            //Act
            ControlPropertySetter.Set(propClass, props);

            //Assert
            var output = new Output();
            
            output.WrapLine("Property set:");
            output.WriteLine();
            output.FormatTable(props.Select(p => new { p.Property, p.Value }));
            output.WriteLine();
            output.WriteLine();
            output.WrapLine("Extracted:");
            output.WriteLine();
            output.FormatTable(new [] { propClass });
            output.Report.Verify();
        }
    }
}