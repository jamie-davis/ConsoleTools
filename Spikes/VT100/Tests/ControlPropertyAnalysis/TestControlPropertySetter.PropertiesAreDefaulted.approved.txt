Property set:                                                                                  
                                                                                               
Property          Value                                                                        
---------- ------------                                                                        
Background        Black                                                                        
Foreground        White                                                                        
                                                                                               
                                                                                               
Extracted:                                                                                     
                                                                                               
      Button       Button                                                                      
  Background   Foreground                                                                      
------------ ------------                                                                      
       Black        White                                                                      
                                                                                               